packages/ckeditor5-table/src/tableutils.js

```diff
@@ -116,12 +116,17 @@ export default class TableUtils extends Plugin {
 	 * @param {Object} options
 	 * @param {Number} [options.at=0] The row index at which the rows will be inserted.
 	 * @param {Number} [options.rows=1] The number of rows to insert.
+	 * @param {Number} [options.copyStructureFrom] The row index used as structure reference.
 	 */
 	insertRows( table, options = {} ) {
 		const model = this.editor.model;
 
 		const insertAt = options.at || 0;
 		const rowsToInsert = options.rows || 1;
+		const copyStructureFrom = options.copyStructureFrom !== undefined ? options.copyStructureFrom : -1;
```

Comment:
I would like to make this more digestible. Using `-1` as an indicator for something looks like a smart shortcut. I'm worried here that using magic number is harder to analyse why some comparison are made in such way.

I think that a switch `isCopyStructure = copyStructureFrom !== undefined` would be simpler to follow.

The second thing I think it would be better to have is to revert the additions to `createEmptyRows()` to it previous form. And use calculated spans directly to create empty cells (using `createEmptyCell()` ). This way we shouldn't have to create two arrays for that same information (`columnSpans` and `cellsAttributes`) and can use one of them directly.